cpumanager should graceful exit other than panic when met error

**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:
When enable cpumanager by `--cpu-manager-policy=static`,  but not set `--kube-reserved` and `--system-reserved`, kubelet will get panic error in log
```
I1214 03:22:51.865756   23951 cpu_manager.go:120] [cpumanager] detected CPU topology: &{2 2 1 map[0:{0 0} 1:{0 1}]}
panic: [cpumanager] unable to determine reserved CPU resources for static policy

goroutine 88 [running]:
k8s.io/kubernetes/pkg/kubelet/cm/cpumanager.NewManager(0x7fff04c660c7, 0x6, 0x2540be400, 0xc420b96738, 0xc421442360, 0x497d024, 0x10, 0xffffffffffffffff, 0x0, 0x80, ...)
        /data/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/cm/cpumanager/cpu_manager.go:124 +0x65b
k8s.io/kubernetes/pkg/kubelet/cm.NewContainerManager(0xa682480, 0xc420c17160, 0xa680780, 0xc4214109c0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /data/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/cm/container_manager_linux.go:278 +0xb0e
k8s.io/kubernetes/cmd/kubelet/app.run(0xc420496600, 0xc420ef1e40, 0xc42003be98, 0x1)
        /data/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/kubelet/app/server.go:454 +0xc22
k8s.io/kubernetes/cmd/kubelet/app.Run(0xc420496600, 0x0, 0x10, 0x43dd620)
        /data/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/kubelet/app/server.go:183 +0xfa
main.NewKubelet.func1(0xc4208697a0, 0xc420c28000, 0x0, 0x1b, 0xc420124600, 0x1, 0x10000a9f0280)
        /data/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/hyperkube/kubelet.go:45 +0x75
main.(*HyperKube).Run.func1(0xc4201245a0, 0xc4208697a0)
        /data/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/hyperkube/hyperkube.go:216 +0x6d
created by main.(*HyperKube).Run
        /data/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/hyperkube/hyperkube.go:215 +0x733

```
**What you expected to happen**:
Should graceful exit with error not panic.

**How to reproduce it (as minimally and precisely as possible)**:
set `--cpu-manager-policy=static` and but not set `--kube-reserved` and `--system-reserved`

**Anything else we need to know?**:

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/cpumanager/cpu_manager.go#L122-L135

--- a/pkg/kubelet/cm/cpumanager/cpu_manager.go
+++ b/pkg/kubelet/cm/cpumanager/cpu_manager.go
@@ -98,13 +98,7 @@ type manager struct {
 var _ Manager = &manager{}
 
 // NewManager creates new cpu manager based on provided policy
-func NewManager(
-	cpuPolicyName string,
-	reconcilePeriod time.Duration,
-	machineInfo *cadvisorapi.MachineInfo,
-	nodeAllocatableReservation v1.ResourceList,
-	stateFileDirecory string,
-) (Manager, error) {
+func NewManager(cpuPolicyName string, reconcilePeriod time.Duration, machineInfo *cadvisorapi.MachineInfo, nodeAllocatableReservation v1.ResourceList, stateFileDirecory string) (Manager, error) {
 	var policy Policy
 
 	switch policyName(cpuPolicyName) {
@@ -120,18 +114,16 @@ func NewManager(
 		glog.Infof("[cpumanager] detected CPU topology: %v", topo)
 		reservedCPUs, ok := nodeAllocatableReservation[v1.ResourceCPU]
 		if !ok {
-			// The static policy cannot initialize without this information. Panic!
-			panic("[cpumanager] unable to determine reserved CPU resources for static policy")
+			// The static policy cannot initialize without this information.
+			return nil, fmt.Errorf("[cpumanager] unable to determine reserved CPU resources for static policy")
 		}
 		if reservedCPUs.IsZero() {
-			// Panic!
-			//
 			// The static policy requires this to be nonzero. Zero CPU reservation
 			// would allow the shared pool to be completely exhausted. At that point
 			// either we would violate our guarantee of exclusivity or need to evict
 			// any pod that has at least one container that requires zero CPUs.
 			// See the comments in policy_static.go for more details.
-			panic("[cpumanager] the static policy requires systemreserved.cpu + kubereserved.cpu to be greater than zero")
+			return nil, fmt.Errorf("[cpumanager] the static policy requires systemreserved.cpu + kubereserved.cpu to be greater than zero")
 		}
 
 		// Take the ceiling of the reservation, since fractional CPUs cannot be

--- a/pkg/kubelet/cm/cpumanager/cpu_manager_test.go
+++ b/pkg/kubelet/cm/cpumanager/cpu_manager_test.go
@@ -234,7 +234,6 @@ func TestCPUManagerGenerate(t *testing.T) {
 		cpuPolicyName              string
 		nodeAllocatableReservation v1.ResourceList
 		isTopologyBroken           bool
-		panicMsg                   string
 		expectedPolicy             string
 		expectedError              error
 		skipIfPermissionsError     bool
@@ -270,14 +269,14 @@ func TestCPUManagerGenerate(t *testing.T) {
 			description:                "static policy - broken reservation",
 			cpuPolicyName:              "static",
 			nodeAllocatableReservation: v1.ResourceList{},
-			panicMsg:                   "unable to determine reserved CPU resources for static policy",
+			expectedError:              fmt.Errorf("unable to determine reserved CPU resources for static policy"),
 			skipIfPermissionsError:     true,
 		},
 		{
 			description:                "static policy - no CPU resources",
 			cpuPolicyName:              "static",
 			nodeAllocatableReservation: v1.ResourceList{v1.ResourceCPU: *resource.NewQuantity(0, resource.DecimalSI)},
-			panicMsg:                   "the static policy requires systemreserved.cpu + kubereserved.cpu to be greater than zero",
+			expectedError:              fmt.Errorf("the static policy requires systemreserved.cpu + kubereserved.cpu to be greater than zero"),
 			skipIfPermissionsError:     true,
 		},
 	}
@@ -319,19 +318,6 @@ func TestCPUManagerGenerate(t *testing.T) {
 				t.Errorf("cannot create state file: %s", err.Error())
 			}
 			defer os.RemoveAll(sDir)
-			defer func() {
-				if err := recover(); err != nil {
-					if testCase.panicMsg != "" {
-						if !strings.Contains(err.(string), testCase.panicMsg) {
-							t.Errorf("Unexpected panic message. Have: %q wants %q", err, testCase.panicMsg)
-						}
-					} else {
-						t.Errorf("Unexpected panic: %q", err)
-					}
-				} else if testCase.panicMsg != "" {
-					t.Error("Expected panic hasn't been raised")
-				}
-			}()
 
 			mgr, err := NewManager(testCase.cpuPolicyName, 5*time.Second, machineInfo, testCase.nodeAllocatableReservation, sDir)
 			if testCase.expectedError != nil {

